### PR TITLE
Refactor singularity version checks into dedicated function

### PR DESCRIPTION
**Pull request by jnc74743**

### CODE REVIEW

**Review by jose-caballero — COMMENTED** [review]
Maybe it is cleaner if these lines

    # For older Singularity versions we must define Docker capabilities.
    # Newer Singularity versions rely on security-options
    # This checks which image we are using and sets Docker create options accordingly

also go inside the function get_singularity_major_version(argv)

**Comment by jnc74743** [comment]
Good call, that does look cleaner

**Review by jose-caballero — APPROVED** [review]
OK

File: docker.py [hunk 1]
```diff
@@ -13,10 +13,20 @@ def gateway():
         gateway_needed = True
     return gateway_needed and not os.path.isfile('/etc/nogateway')
 
-def args_create(argv):
-    # Define major Singularity version
-    singularity_major_version = int(3)
+def get_singularity_major_version(argv):
+    # For older Singularity versions we must define Docker capabilities.
+    # Newer Singularity versions rely on security-options
+    # This checks which image we are using and sets Docker create options accordingly
```
Comment (jrha) [inline docker.py:19]: But this function doesn't set any options…

Comment (jrha) [reply docker.py:19]: It _allows_ us to set them

**Pull request closed without merging**